src/js/components/Search/Search.tsx

```diff
@@ -5,29 +5,80 @@ import { useTranslation } from 'react-i18next';
 import SearchFieldsStack from './SearchFieldsStack';
 import SearchResults from './SearchResults';
 
-import { makeGetKatsuPublic } from '@/features/search/query.store';
+import { makeGetKatsuPublic, setQueryParams } from '@/features/search/query.store';
 import { NON_DEFAULT_TRANSLATION } from '@/constants/configConstants';
 import { useAppDispatch, useAppSelector } from '@/hooks';
+import { useLocation, useNavigate } from 'react-router-dom';
 
-const Search = () => {
-  const { t } = useTranslation(NON_DEFAULT_TRANSLATION);
+type QueryParams = { [key: string]: string };
+
+const SearchRouter: React.FC = () => {
```

Comment:
this is routed but it isn't a router - call it `SearchRouted` or `RoutedSearch`, whichever ur pref